components/properties/src/provider.rs

```diff
@@ -139,76 +141,114 @@ pub mod key {
-        UnicodePropertyV1 {
+    fn default() -> CodePointSet<'static> {
+        CodePointSet {
             inv_list: UnicodeSetBuilder::new().build(),
         }
     }
 }
 
-impl<'data> UnicodePropertyV1<'data> {
-    /// Creates a [`UnicodePropertyV1`] for the given [`UnicodeSet`].
-    pub fn from_owned_uniset(set: UnicodeSet<'data>) -> UnicodePropertyV1<'data> {
-        UnicodePropertyV1 { inv_list: set }
+impl<'data> CodePointSet<'data> {
+    /// Creates a [`UnicodeProperty`] for the given [`UnicodeSet`].
+    pub fn from_owned_uniset(set: UnicodeSet<'data>) -> CodePointSet<'data> {
+        CodePointSet { inv_list: set }
     }
 }
 
-impl<'data> From<UnicodePropertyV1<'data>> for UnicodeSet<'data> {
-    fn from(prop: UnicodePropertyV1<'data>) -> UnicodeSet<'data> {
+impl<'data> From<CodePointSet<'data>> for UnicodeSet<'data> {
+    fn from(prop: CodePointSet<'data>) -> UnicodeSet<'data> {
         prop.inv_list
     }
 }
 
-//
-// UnicodePropertyMap
-//
+/// A set of characters with a particular property, loaded from data
+#[derive(Clone)]
+pub struct CodePointSetData<T: CodePointSetLikeProperty> {
+    pub(crate) data: DataPayload<CodePointSetMarker<T>>,
+}
 
-/// A map efficiently storing data about individual characters.
+impl<T: CodePointSetLikeProperty> CodePointSetData<T> {
+    pub fn contains(&self, ch: char) -> bool {
+        self.data.get().inv_list.contains(ch)
+    }
+}
+
+/// Internal data type for [`CodePointMapData`]
 #[derive(Debug, Eq, PartialEq, yoke::Yokeable, zerofrom::ZeroFrom)]
 #[cfg_attr(feature = "datagen", derive(serde::Serialize))]
 #[cfg_attr(feature = "serde", derive(serde::Deserialize))]
-pub struct UnicodePropertyMapV1<'data, T: TrieValue> {
+pub struct CodePointMap<'data, T: TrieValue> {
     /// A codepoint trie storing the data
     #[cfg_attr(feature = "serde", serde(borrow))]
     pub code_point_trie: CodePointTrie<'data, T>,
 }
 
-impl<'data, T: TrieValue> Clone for UnicodePropertyMapV1<'data, T>
+impl<'data, T: TrieValue> Clone for CodePointMap<'data, T>
 where
     <T as zerovec::ule::AsULE>::ULE: Clone,
 {
     fn clone(&self) -> Self {
-        UnicodePropertyMapV1 {
+        CodePointMap {
             code_point_trie: self.code_point_trie.clone(),
         }
     }
 }
 
-/// Marker type for UnicodePropertyMapV1.
-/// This is generated by hand because icu_provider::data_struct doesn't support generics yet.
-pub struct UnicodePropertyMapV1Marker<T: TrieValue> {
-    _phantom: core::marker::PhantomData<T>,
+/// A map of characters to property values, loaded from data
+#[derive(Clone)]
+pub struct CodePointMapData<T: CodePointMapLikeProperty> {
+    pub(crate) data: DataPayload<CodePointMapMarker<T>>,
+}
+
+impl<T: CodePointMapLikeProperty> CodePointMapData<T> {
+    pub fn get(&self, ch: char) -> T::Value {
+        self.data.get().code_point_trie.get(ch as u32)
+    }
+}
+
+/// Properties that are stored in CodePointSets
+pub trait CodePointSetLikeProperty {
```

Comment:
n.b. this and the other trait should just be CodePointSetProperty

CPSLike/etc haven't been introduced yet